extensions/git/src/repository.ts

```diff
@@ -855,12 +855,23 @@ export class Repository implements Disposable {
 		globalState: Memento,
 		outputChannel: OutputChannel
 	) {
-		const workspaceWatcher = workspace.createFileSystemWatcher('**');
+		const workspaceWatcher = workspace.createFileSystemWatcher(new RelativePattern(repository.root, '**'));
```

Comment:
Suggest `new RelativePattern(Uri.file(repository.root),` to use a `URI`

Reply:
Would `new RelativePattern(repository.root, '**')` detect if for some reason one deletes `repository.root`?

Reply:
The API docs of `createFileSystemWatcher` try to explain that:

https://github.com/microsoft/vscode/blob/f5ad06212dad13272d0de2451b803b30bb5ef831/src/vscode-dts/vscode.d.ts#L10986-L10990

Reply:
So, `repository.root` is not the workspace folder? 

Btw, I should maybe clarify that comment: any folder that is inside the workspace will be watched for deletion because we setup recursive watching for every folder that is within the workspace, including all `.git` folders. Only when you add a path that is not yet watched, you will face this.

Reply:
Updated the JSDoc to clarify the behaviour:

https://github.com/microsoft/vscode/blob/21fae2cd4632ea19e3eb38dbb2dbb483bd6388e4/src/vscode-dts/vscode.d.ts#L10986-L10997

Reply:
> So, `repository.root` is not the workspace folder?

Not necessarily. You can open a folder/workspace that has one/more git repositories further down the directory structure.

Reply:
So do we need to do something here?

Reply:
> Suggest `new RelativePattern(Uri.file(repository.root),` to use a `URI`

I did push a change to switch to using the Uri